### DISCUSSION TITLE
Coupling NekRS with external CUDA code.

### DISCUSSION THREAD

**Discussion by juanecopro** [Nek5000/nekRS#283] [Compiling] [answered]
Hi All,
I'm trying to create a plugin which will use external code that uses CUDA. What would be the correct way to do this?

I was able to compile when using Serial backend, but I'm having trouble deciphering who to compile the code that links to an external library that uses CUDA. Do I need to set the main compiler to NVCC??

Thanks,
Juan Diego

**Comment by stgeke** [comment, marked as answer]
Please contact the occa team to discuss this question.
